package/MDAnalysis/transformations/translate.py

```diff
@@ -63,8 +65,75 @@ def wrapped(ts):
             v = np.float32(vector)
             ts.positions += v
         else:
-            raise ValueError("{} vector in too short".format(vector))
+            raise ValueError("{} vector is too short".format(vector))
+        
+        return ts
+    
+    return wrapped
+
+def center_in_box(ag, center='geometry', point=None, pbc=None):
+    """
+    Translates the coordinates of a given :class:`~MDAnalysis.coordinates.base.Timestep`
+    instance so that the center of geometry/mass of the given :class:`~MDAnalysis.core.groups.AtomGroup`
+    is centered on the unit cell. The unit cell dimensions are taken from the input Timestep object or
+    can be defined using the `box` argument.
+    
+    Example
+    -------
+    
+    .. code-block:: python
+    
+        ag = u.residues[1].atoms
+        ts = MDAnalysis.transformations.center(ag,center='mass')(ts)
+    
+    Parameters
+    ----------
+    ag: AtomGroup
+        atom group to be centered on the unit cell.
+    center: str, optional
+        used to choose the method of centering on the given atom group. Can be 'geometry'
+        or 'mass'
+    point: array-like, optional
+        overrides the unit cell center - the coordinates of the Timestep are translated so
+        that the center of mass/geometry of the given AtomGroup is aligned to this position
+        instead. Defined as an array of size 3. Example: [1, 2, 3]
+    pbc: bool or None, optional
+        If True, all the atoms from the given AtomGroup will be moved to the unit cell
+        before calculating the center of mass or geometry
+    
+    Returns
+    -------
+    :class:`~MDAnalysis.coordinates.base.Timestep` object
+    """
+    
+    pbc_arg = pbc
+    try:
+        if center == 'geometry':
+            center_method = partial(ag.center_of_geometry, pbc=pbc_arg)
+        elif center == 'mass':
+            center_method = partial(ag.center_of_mass, pbc=pbc_arg)
+        else:
+            raise ValueError('{} is not a valid argument for center'.format(center))
+    except AttributeError:
+        if center == 'mass':
+            raise AttributeError('{} is not an AtomGroup object with masses'.format(ag))
+        else:
+            raise ValueError('{} is not an AtomGroup object'.format(ag))
+
+    def wrapped(ts):
+        ag_center = center_method()
+        if point:
+            if len(point)==3:
```

Comment:
The comment disappeared, but this remains to change: you are testing and converting the point to an array for each frame while it could be done once in the parent function.